addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -57,7 +57,10 @@ function factory(dependencies) {
          * Send the attachment for the browser to download.
          */
         download() {
-            this.env.services.navigate(`/web/content/ir.attachment/${this.id}/datas`, { download: true });
+            let downloadLink = document.createElement('a');
+            downloadLink.setAttribute('href', `/web/content/ir.attachment/${this.id}/datas`);
+            downloadLink.setAttribute('download','');
```

Comment:
Hi @zel-odoo  @seb-odoo , why does the attachment (link) not open a new tab. I find the user experience very annoying, for example I am in Project > Task > click on the attachment > if I open the current tab, the breadcrumbs (previous filters) will be lost.

How do I reproduce: interrupting the activity in the page such as rtc call

Reply:
@zel-odoo can you have a look?

Reply:
@zel-odoo ping

Reply:
hello @royle-vietnam , I dont really get the problem, can you maybe record it or describe it again? What do you mean by saying "attachment (link) not open a new tab"? The code you pointed out was only for downloading the attachment so I think it will go to the download manager directly if I understand correctly

![Uploading Peek 2024-10-15 10-54.gif…]()

Reply:
Hi @zel-odoo , for link-type attachments, when clicking the download button, I would like to open a new tab to avoid losing breadcrumbs (or previous filters) instead of opening the link in the current tab.
![Screenshot from 2024-10-15 16-04-53](https://github.com/user-attachments/assets/c0bc8ff8-8cff-4aba-973b-ff225f318b0a)

Reply:
ok, I see what you mean. @aju-odoo , for me, downloading a "link" is a little bit weird. What the fix can do is
1. do not change the download button and when the file type is a link, we always open in a new tab;
2. when the file type is a link, the download button changes into an "open" button and the user can choose to open in the same tab (by default) or "open in a new tab" with a right click, as the normal behavior we have in Odoo.

Reply:
@zel-odoo I would go for option 2, but open the link in a new tab by default so that users do not lose the record they were currently viewing.

Reply:
@seb-odoo @aju-odoo @royle-vietnam https://github.com/odoo/odoo/pull/184264